pkg/cloudprovider/providers/azure/azure_blob.go

```diff
@@ -41,8 +42,19 @@ func (az *Cloud) createVhdBlob(accountName, accountKey, name string, sizeGB int6
 	name = name + ".vhd"
 	err = blobClient.PutPageBlob(vhdContainerName, name, vhdSize, tags)
 	if err != nil {
+		// if container doesn't exist, create one and retry PutPageBlob
```

Comment:
any reason you do this rather than just querying for the storage container to see if it exists, rather than handling it as an error?